### ISSUE TITLE
s3bucket_is_ok incompatible with cross account setups

### ISSUE THREAD

**Issue by kpcyrd** [shinsaka/fluent-plugin-elb-log#26]
I got the following error:

```
2017-08-01 17:16:12 +0000 [error]: unexpected error error_class=StandardError error=#<StandardError: s3 bucket not found xxxx-redacted>
```

but I've verified the bucket exists and is accessible for my access key. I digged into the code and it appears this exception is caused here

```
 raise StandardError.new("s3 bucket not found #{@s3_bucketname}") unless s3bucket_is_ok()
```

and the bug being here:

```
  def s3bucket_is_ok
    begin
      log.debug "search bucket #{@s3_bucketname}"

      s3_client.list_buckets.buckets.any? do |bucket|
        bucket.name == @s3_bucketname
      end
    rescue => e
      log.warn "S3 Client error occurred: #{e.message}"
    end
end
```

I'm not a ruby programmer, but I think this is basically doing `aws s3api list-buckets` and searching for the bucket that way, which doesn't work if I'm trying to access a bucket in a different aws account. I'd recommend to remove this check and just raise an exception if the bucket is actually not accessible.

I'm on the current stable fluentd (0.12) and the latest fluent-plugin-elb-log that still supports this version (0.2.8).

**Comment by shinsaka** [comment]
Thank you for your report.
I'll try to make sure.

**Comment by shinsaka** [comment]
@kpcyrd 
I support for S3 with cross account setups,and I released gem version 0.4.1.
Could you try new version.

Thank you for your report.
